### ISSUE TITLE
Mettre les données dans OSM

### ISSUE THREAD

**Issue by laem** [benoitdemaegdt/voieslyonnaises#148]
Bonjour @benoitdemaegdt, d'abord merci pour ce projet ! Le site est propre, la carte est très lisible. 

J'étais curieux de voir dans quel format tu stockais les données. Je me demande : pourquoi ne pas reposer sur OSM ? 

Actuellement dans OSM, les voies cyclables de qualité de type Voies Lyonnaises ou REV à Rennes ne sont pas tagguées comme faisant partie de ce réseau particulier. 

Un bel exemple ici, l'une des rares portions terminées des VL (d'après un excellent site, cyclopolis.fr). 

https://www.openstreetmap.org/way/867993162#map=17/45.74136/4.84947

On voit qu'elle n'a aucuns tags permettant de comprendre qu'elle fera partie d'un ensemble plus important. 
![image](https://github.com/benoitdemaegdt/voieslyonnaises/assets/1177762/68f8ed1f-8855-4973-b8d9-c9870b140963)

Pourtant, il me semble faisable et courant d'ajouter sur OSM des relations entre les `way` qui permettent de donner du sens à une échelle plus grande. 

Exemple ici sur les berges du Rhone, où la voie cyclable fait partie de deux relations, ` Voie Verte des Berges du Rhône à Lyon ` et `[ViaRhôna 10: Lyon - Saint-Romain-en-Gal/Vienne](https://www.openstreetmap.org/relation/10792982)`.
https://www.openstreetmap.org/way/172814295

On pourrait donc créer une relation "Voie Lyonnaise n°1". 

Quid des segments non terminés ou pas encore commencés ? 

Ici à Rennes, ce segment du REV est marqué comme "en construction", et auparavant était noté "proposé". Je viens de faire la MAJ pour le transformer en "en construction". 

![image](https://github.com/benoitdemaegdt/voieslyonnaises/assets/1177762/a1ebea10-08cc-43e4-969e-ec11d4deae78)

Qu'en penses-tu ?

**Comment by benoitdemaegdt** [comment]
Salut @laem 👋 

C'est assez marrant que tu me contactes ici, parce que d'une certaine manière, tu es à l'origine de ce projet. C'est en effet suite à un de tes tweets sur le réseau cyclable lyonnais, qui doit dater d'il y a un peu plus d'un an, que j'ai commencé à mettre sur une carte les premières sections des voies lyonnaises 😄 

Allez, place aux réponses : 

> J'étais curieux de voir dans quel format tu stockais les données

J'utilise des fichiers geojson, 1 fichier par voie lyonnaise, soit 12 fichiers au total. Chaque fichier est découpé en tronçons. Le découpage est pour l'instant arbitraire, avec un objectif moyen terme qu'il corresponde au découpage des phases de travaux (pour faciliter le changement de statut : "prévu" -> "en travaux" -> "terminé"). Les tronçons sont générés à la main via le site geojson.io.

> pourquoi ne pas reposer sur OSM ?

Il n'y a pas vraiment de raison. Au début du projet, je lisais les dossiers de concertation, je repérais les rues dans lesquelles passeront les voies lyonnaises, et je créais le geojson associé. C'était ensuite très facile d'intégrer ces fichiers geojson dans une webapp, parce que j'en ai déjà une autre qui tourne sur laquelle j'avais déjà intégré des geojson (ex : https://monpetitsommet.fr/randonnees/petite-traversee-chartreuse). C'était donc la solution la plus rapide, très bien pour un side project.

Aujourd'hui le projet prend un peu d'ampleur, et la question de la gestion des données peut se reposer. Je ne connais pas l'univers OSM, et donc pas trop d'avis sur les points que tu soulèves.
Je suis par contre favorable à rendre cette donnée plus facilement accessible. J'ai un item dans ma todo liste qui s'appelle : "déposer les fichiers geojson sur data.gouv". Ce serait probablement mieux de mettre directement sur OSM 😃 

Je vais finir le projet avec le système en place (ETA fin du projet début octobre 2023). Si jamais tu as un peu de dispo, je serais chaud pour que tu m'expliques ensuite les grandes lignes d'une intégration avec OSM 😃

**Comment by laem** [comment]
Je viens de découvrir ce sujet : https://forum.openstreetmap.fr/t/numerisation-des-reseaux-velo-structurants/17214

**Comment by laem** [comment]
Salut @benoitdemaegdt. 

Après réflexion, je me dis que ton format n'est pas mal du tout pour commencer. Et qu'à partir de ça, il y a sûrement moyen facilement d'injecter les données dans OSM. 

L'immense avantage, c'est qu'il permet aux villes qui n'ont rien prévu d'inventer des schémas cyclables, qu'on ne pourrait pas mettre sur OSM tant que c'est des propositions politiques. 

Du coup j'ai intégré tes données sur cette démo de villes.plus https://villes-plus-ntm870dzl-menoz.vercel.app/cyclables/Métropole de Lyon

Rien de public pour l'instant, je mettrai les mentions. Ça fait suite à une discussion sur le mattermost de la FUB où une asso veut créer son "plan style métro" d'une proposition de REV structurant.

**Comment by benoitdemaegdt** [comment]
Hello @laem  👋 

> Après réflexion, je me dis que ton format n'est pas mal du tout pour commencer. Et qu'à partir de ça, il y a sûrement moyen facilement d'injecter les données dans OSM.

Oui, je n'ai toujours pas regardé, mais le geojson est un format assez standard. ça ne devrait pas être bien compliqué de le passer sur OSM le jour ou on souhaite le faire 🤞 

> L'immense avantage, c'est qu'il permet aux villes qui n'ont rien prévu d'inventer des schémas cyclables, qu'on ne pourrait pas mettre sur OSM tant que c'est des propositions politiques.

En effet, je n'y avais pas pensé, mais c'est un bon point ! 👍 

> Du coup j'ai intégré tes données sur cette démo de villes.plus [https://villes-plus-ntm870dzl-menoz.vercel.app/cyclables/Métropole](https://villes-plus-ntm870dzl-menoz.vercel.app/cyclables/M%C3%A9tropole) de Lyon
Rien de public pour l'instant, je mettrai les mentions. Ça fait suite à une discussion sur le mattermost de la FUB où une asso veut créer son "plan style métro" d'une proposition de REV structurant.

Très cool, si d'autres asso peuvent reprendre ce système pour pousser vers d'autres aménagements, c'est juste le top 🎉 
Tu verras que dans chaque "features" des geojson, il y a un objet "properties", qui contient une clé "status". ça permet de différencier les différents états : prévu, abandonné, en travaux, variante envisagée ... Il me semble que sur ta carte tu as tout ajouté, mais malheureusement ce ne sera pas la tête des voies lyonnaises en 2026 😓 
Enfin pour un démonstrateur pour d'autres asso, c'est déjà top !

**Comment by laem** [comment]
>  Il me semble que sur ta carte tu as tout ajouté, mais malheureusement ce ne sera pas la tête des voies lyonnaises en 2026

Oui, tout à fait. Il faut que je clarifie l'UI, mais l'idée c'est de se projeter dans le futur, après le constat accablant du moment :)

**Comment by benoitdemaegdt** [comment]
> où une asso veut créer son "plan style métro" d'une proposition de REV structurant

Reste le soucis de la création des fichiers geojson à proprement parler.
Ici on passe par [geojson.io](http://geojson.io/#map=2/0/20), ça fonctionne bien, mais c'est laborieux (il faut faire le tracé point par point).
Idéalement il faudrait trouver un outil similaire à ce qui existe pour les apps de rando, qui 'aimante' à la route la plus proche.

**Comment by hverlin** [comment]
Il y a maintenant un article à ce sujet, avec une carte `umap` qui compare les données OSM avec cyclopolis.

https://cyclopolis.fr/blog/voies-lyonnaises-open-street-map#diff%C3%A9rences-entre-openstreetmap-et-cyclopolis
